hoomd/hpmc/field.py

```diff
@@ -32,208 +37,136 @@ def __init__(self):
         # nothing else to do.
 
 
-class lattice_field(_external):
-    R""" Restrain particles on a lattice
+class ExternalField(_HOOMDBaseObject):
+    """Base class external field.
 
-    Args:
-        mc (:py:mod:`hoomd.hpmc.integrate`): MC integrator.
-        position (list): list of positions to restrain each particle (distance units).
-        orientation (list): list of orientations to restrain each particle (quaternions).
-        k (float): translational spring constant.
-        q (float): rotational spring constant.
-        symmetry (list): list of equivalent quaternions for the shape.
-        composite (bool): Set this to True when this field is part of a :py:class:`external_field_composite`.
-
-    :py:class:`lattice_field` specifies that a harmonic spring is added to every particle:
-
-    .. math::
-
-        V_{i}(r)  = k_r*(r_i-r_{oi})^2 \\
-        V_{i}(q)  = k_q*(q_i-q_{oi})^2
+    Provides common methods for all subclasses.
 
     Note:
-        1/2 is not included in the formulas, specify your spring constants accordingly.
+        Users should use the subclasses and not instantiate `ExternalField`
+        directly.
+    """
 
-    * :math:`k_r` - translational spring constant.
-    * :math:`r_{o}` - lattice positions (in distance units).
-    * :math:`k_q` - rotational spring constant.
-    * :math:`q_{o}` - lattice orientations (quaternion)
 
-    Once initialized, the compute provides the following log quantities that can be logged via analyze.log:
+class LatticeField(ExternalField):
+    R"""Restrain particles on a lattice
```

Comment:
Just to be clear, this docstring needs the `r"""` because of the backslashes in the equations, right?

Reply:
Perhaps, but not necessarily. In may places in the docs, I need to use `"""` and escape every backslash so that the single backslash character can be used for line continuation to correctly format sphinx docs while meeting the flake8 checks.